### PR TITLE
New Functions article and related changes.

### PR DESCRIPTION
**Pull request by sbudhirajadoc**
New functions article plus an extensions update.

### What does this PR do?
Creates a new article for Salesforce Functions related changes to VSCode

### What issues does this PR fix or reference?
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000NPt6YAG/view

### Functionality Before
Could not create, start, stop or debug a Saleforce Function in VSCode.

### Functionality After
We can now create, start, stop or debug a Saleforce Function in VSCode.

### CODE REVIEW

File: docs/_articles/en/functions/overview.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+title: Overview
+lang: en
+---
+## Salesforce Functions (Beta)
+Use the Salesforce Functions extensions in VSCode to build event-driven, elastically scalable apps and experiences.
+
+Use the Salesforce features for VS Code to create and test Salesforce Functions locally.
```
Comment (ekapner) [inline docs/_articles/en/functions/overview.md:8]: If this sentence is intended to be a lead-in to the head3s, then I would edit it to say something like:
Use these VS Code features to create and test... But I'm not sure this extra sentence adds any real value that the first sentence doesn't already cover.

File: docs/_articles/en/functions/overview.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Create Function
+Run `SFDX: Create Function` to create a Salesforce Function in Javascript or Java in a Salesforce DX project.
+
+A new directory named after your Function name and all the supporting files are created for you.
```
Comment (ekapner) [inline docs/_articles/en/functions/overview.md:13]:
```suggestion
This command creates a new directory named after your Function name and all the supporting files.
```

File: docs/_articles/en/functions/overview.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+* VS Code creates the basic scaffolding that contains files with some rudimentary, boilerplate code and supporting metadata.
+* The `package.json` file contains information about dependencies.
+
+![Create Function](../../../images/vs_code_create_func.gif)
+### Start Function
+Run `SFDX: Start Function` from the command palette to run the function locally.
```
Comment (ekapner) [inline docs/_articles/en/functions/overview.md:20]: You don't say "from the command palette" in Create Function. Do we think that's implied at this point? That all the VS code commands are "from" the palette.

File: docs/_articles/en/functions/overview.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+### Invoke Function
+The Salesforce extensions add `Invoke` and `Debug Invoke` CodeLens to a file that is in the correct format to send test events to a Function that is running locally. With one click, you can test a function that is running, by invoking it with a mock payload. You can also debug the function against the payload.
```
Comment (ekapner) [inline docs/_articles/en/functions/overview.md:23]: CodeLens(es)? or add "the" Invoke Debug Invoke CodeLens...

File: docs/_articles/en/functions/overview.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+A sample `payload.json` file that can invoke a function:
+
+![Invoke Debug ](../../../images/vscode_func_payload.png)
+
+### Stop Function
+Stop a function by running `SFDX: Stop Function` from the command palette. This operation simply kills the local process.
```
Comment (ekapner) [inline docs/_articles/en/functions/overview.md:30]:
```suggestion
Stop a function by running `SFDX: Stop Function`, which simply kills the local process.
```

File: docs/_articles/en/getting-started/recommended-extensions.md [hunk 6]
```diff
@@ -40,3 +40,6 @@ VS Code doesn’t ship with rich XML tools by default. This extension, built by
 ## Community Extensions
 
 The community has developed several extensions to help make Salesforce development more productive. To find these, search the [Visual Studio Code Marketplace](https://marketplace.visualstudio.com/vscode) or browse all extensions tagged [Salesforce](https://marketplace.visualstudio.com/search?term=tag%3Asalesforce&target=VSCode&category=All%20categories&sortBy=Relevance).
+
+## Debugger for Java
+Java Debugger based on Java Debug Server which extends the Language Support for Java by Red Hat. It allows users to debug Java code in VSCode. Install this extension to invoke and debug Java functions.
```
Comment (ekapner) [inline docs/_articles/en/getting-started/recommended-extensions.md:45]:
```suggestion
Java Debugger based on Java Debug Server that extends the Language Support for Java by Red Hat. It allows you to debug Java code in VSCode. Install this extension to invoke and debug Java functions.
```

**Review by ekapner — COMMENTED** [review]
A couple editorial comments.

File: docs/_articles/en/functions/overview.md (hunk 1 above)
Comment (sbudhirajadoc) [reply docs/_articles/en/functions/overview.md:8]: How is this:
Use the Salesforce Functions features in VSCode to build event-driven, elastically scalable apps and experiences. You can create and test functions against sample payloads locally using the VSCode IDE.  @ekapner

Comment (ekapner) [reply docs/_articles/en/functions/overview.md:8]: @sbudhirajadoc  - one small tweak. 

Use the Salesforce Functions features in VSCode to build event-driven, elastically scalable apps and experiences. You can create and test functions against sample payloads locally in VS Code.

Comment (ekapner) [reply docs/_articles/en/functions/overview.md:8]: I think there's a space between VS and Code - I looked at MS docs and they include a space.

Comment (sbudhirajadoc) [reply docs/_articles/en/functions/overview.md]: Got it. Thanks Emily!

**Review by ekapner — APPROVED** [review]

File: docs/_articles/en/functions/overview.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+* The `package.json` file contains information about dependencies.
+
+![Create Function](../../../images/vs_code_create_func.gif)
+### Start Function
+Run `SFDX: Start Function` to run the function locally.
+
+### Invoke Function
+The Salesforce extensions add the `Invoke` and `Debug Invoke` CodeLens to a file that is in the correct format to send test events to a Function that is running locally. With one click, you can test a function that is running, by invoking it with a mock payload. You can also debug the function against the payload.
+
+A sample `payload.json` file that can invoke a function:
```
Comment (jag-sfdc) [inline docs/_articles/en/functions/overview.md:23]: Do we want to explicitly call out that users have to create this payload.json file. There was some confusion around this when I was explaining this to a few users.

**Pull request merged**